### ISSUE TITLE
Deno equivalent of nodejs `process.memoryUsage()`?

### ISSUE THREAD

**Issue by Mehuge** [denoland/deno#9980]
https://nodejs.org/api/process.html#process_process_memoryusage

`process.memoryUsage()`

Returns: `<Object>`
```
rss <integer>
heapTotal <integer>
heapUsed <integer>
external <integer>
arrayBuffers <integer>
```

Use case: I am converting backup software written in `nodejs` to `deno` that uses `process.memoryUsage()` to report peak memory usage during the backup, and it would be a useful feature to keep.

**Comment by ry** [comment]
There is a yet unreleased feature called `Deno.core.heapStats()` which provides this information. It should be available in 1.9 which will be released on April 13. 
```
# ./target/release/deno eval --unstable -p 'Deno.core.heapStats()'
{
  totalHeapSize: 3473408,
  totalHeapSizeExecutable: 262144,
  totalPhysicalSize: 2617340,
  totalAvailableSize: 1515774968,
  totalGlobalHandlesSize: 8192,
  usedGlobalHandlesSize: 160,
  usedHeapSize: 2824736,
  heapSizeLimit: 1518338048,
  mallocedMemory: 73860,
  externalMemory: 0,
  peakMallocedMemory: 58200,
  numberOfNativeContexts: 1,
  numberOfDetachedContexts: 0
}
```

**Comment by AaronO** [comment]
@Mehuge @ry I'll submit a PR exposing `Deno.memoryUsage()` later today, like I mentioned previously I don't think the extra fields will be useful to most users so it's best to align with node's `process.memoryUsage()`.
